### PR TITLE
Feature/settings page

### PR DESCRIPTION
**Pull request by mordrax**
A place where connection to the backend can be configured.
Might become a dumping ground for all settings, configs, env, etc...

### CODE REVIEW

File: Eventful/View.elm [hunk 1]
```diff
@@ -13,50 +14,50 @@ import Material.Button as Button
...
-        viewPage = viewFromPage currentPage
-    in
-        viewPage model |> Material.Scheme.top
+    viewHelper model |> Material.Scheme.top
+
 
-viewFromPage : Page -> Model -> Html Msg
-viewFromPage page =
-    case page of
+viewHelper : Model -> Html Msg
```
Comment (JulianLeviston) [inline Eventful/View.elm]: This means this function now has to know about the insides of the model. If you prefer that, that's fine, but it's not as clean IMO.

File: Settings.elm [hunk 2]
```diff
@@ -0,0 +1,60 @@
+module Settings exposing (Settings, Msg, init, view, update)
+
+import String
+import Html exposing (..)
+import Html.Events as HE exposing (..)
+import Html.Attributes as HA exposing (..)
+
```
Comment (JulianLeviston) [inline Settings.elm]: eh! so many splat imports! :(

Comment (JulianLeviston) [inline Settings.elm]: The reason I always bang on about this isn't for us. We both know how to find this stuff out... it's for others in Blake who might want to know how this works. Maybe it's not even a problem, coming from Ruby or Elixir.

**Comment by JulianLeviston** [comment]
:ship:

File: Settings.elm (hunk 2 above)
Comment (mordrax) [inline Settings.elm]: I'm actually explicitly adding the alias in the code as in `HE.onInput` as well, i think this is a good way to code?
And then we can remove the `exposing (..)` altogether and know where the functions are coming from.

Actually in this case, we're using material so those are obsolete

File: Eventful/View.elm (hunk 1 above)
Comment (mordrax) [inline Eventful/View.elm]: yes, the model/update/view should all know about the model. The Main.View needs to know where to route to, and all the code is in the same module anyway right?

File: Eventful/View.elm [hunk 3]
```diff
@@ -13,50 +14,50 @@ import Material.Button as Button
...
+viewHelper ({ currentPage } as model) =
+    case currentPage of
         Index ->
-            indexView
+            indexView model
 
         Settings ->
-            settingsView
+            Settings.view model.settings
+                |> Html.App.map (\msg -> SettingsMsg msg)
```
Comment (JulianLeviston) [inline Eventful/View.elm]: Also, `SettingMsg` is already a function... why wrap it in another function?

File: Eventful/View.elm (hunk 1 above)
Comment (JulianLeviston) [inline Eventful/View.elm]: Sure... it's not as clean though...

IMO the less things that rely on each other the better. Data can pass through things without them having to "understand" how the internals work, just the small piece that they work with... that way, when things are adjusted in code, there are less places for it to have to be adjusted in. Stacking up many pieces like this allows you to make reasonably large impact changes with reasonably small code changes.

It's fine for now, though.

File: Eventful/View.elm (hunk 3 above)
Comment (mordrax) [inline Eventful/View.elm]: Good point, in my defense, i'm just copying cotw code i wrote months ago hehe, will improve!

**Pull request merged**